integrations/extensions/docs/elasticsearch-install-and-setup/watsonx_discovery_install_and_setup.md

```diff
@@ -0,0 +1,430 @@
+    }
+  }'
+  ```
+* `wa-docs` is the index you created when uploading the example file to Elasticsearch cluster. It contains the text data
+* `search_wa-docs` is the search index that has ELSER output field
+* `elser-v2-test` is the ingest pipeline with an inference processor using ELSER model
+
+### Semantic search by using the text_expansion query
+To perform semantic search, use the `text_expansion` query, and provide the query text and the ELSER model ID.
+The example below uses the query text "How to set up custom extension?", the `text_embedding` field contains
+the generated ELSER output:
+  ```bash
+  curl -X GET "${ES_URL}/search-wa-docs/_search?pretty" -u "${ES_USER}:${ES_PASSWORD}" \
+  -H "Content-Type: application/json" --cacert "${ES_CACERT}" -d'
+  {
+     "query":{
+        "text_expansion":{
+           "text_embedding":{
+              "model_id":".elser_model_2",
+              "model_text":"how to set up custom extension?"
+           }
+        }
+     }
+  }'
+  ```
+NOTE: Learn more about [text-expansion-query](https://www.elastic.co/guide/en/elasticsearch/reference/8.11/semantic-search-elser.html#text-expansion-query) from the tutorial.
+
+## Step 3: Build a custom extension in watsonx Assistant for Elasticsearch API
+
+### Provision a watsonx Assistant instance in your CloudPak cluster
+From you CloudPak cluster, you need to provision a watsonx Assistant instance and then create an assistant in the new wxA experience.
+
+### Retrieve Elasticsearch endpoints  
+  The below command will print out the Elasticsearch Cluster IP and port that you will use as the host and port to access Elasticsearch in your assistant on CloudPak.
+  ```shell
+  oc -n ${ES_NAMESPACE} get svc "$ES_CLUSTER-es-http" -o jsonpath='{.spec.clusterIP}:{.spec.ports[0].port}'; echo
+  ```
+
+### Configure TLS connection with watsonx Assistant  
+  By default, watsonx Assistant (wxA) doesn't trust any certificate that's not in its trusted keystore, which means that
+  you won't be able to make TLS connections between wxA and Elasticsearch. To make wxA trust Elastic's self-signed certificate, 
+  here are two options: 
+
+#### Option 1 (**Not Recommended for Production Use**): Enable the `TRUST_ALL_CERTIFICATES` flag in wxA to trust all certificates
+* Define wxA namespace and resource name as environment variables
+  ```shell
+  WA_INSTANCE="wa"
+  WA_NAMESPACE="cpd"
+  ```
+  NOTE: `WA_INSTANCE` is your wxA resource name, and `WA_NAMESPACE` is the namespace where your wxA is deployed.
+
+* Enable the `TRUST_ALL_CERTIFICATES` flag for wxA via a `patch` command
+  ```shell
+  oc -n $WA_NAMESPACE patch wa $WA_INSTANCE --type='merge' -p='{"configOverrides":{"webhooks_connector":{"extra_vars":{"TRUST_ALL_CERTIFICATES":true}}}}'
+  ```
+  Please wait a few minutes for the `wa-webhooks-connector` pod to restart. Once the pod has restarted successfully, 
+  TLS connections between your Elasticsearch service and wxA have been enabled. 
```

Comment:
I would add the following at the end:

> This approach is not recommended for production use, especially for applications that involve sensitive data,  because it introduces a security risk.  It configures wxA to always trust that all services that it connects to through search integrations or custom integrations are the services they expect to be at the IP addresses that the integrations are configured to use.  Option 2 below eliminates this risk by using a certificate to security verify that the Elasticsearch service you are connecting to is the one configured your assistant to connect to.